mlir/lib/Gradient/Transforms/GradMethods/ClassicalJacobian.cpp

```diff
@@ -106,6 +106,80 @@ func::FuncOp genParamCountFunction(PatternRewriter &rewriter, Location loc, func
     return paramCountFn;
 }
 
+func::FuncOp genSplitPreprocessed(PatternRewriter &rewriter, Location loc, func::FuncOp qnode,
```

Comment:
So basically here we split the QNode: qnode= classical(quantum)(...) and only keep the classical part as splitFn?

Reply:
Yup, the split QNode (`"splitpreprocessed"`) is exactly the same as the argmap function except rather than returning the gate parameter vector at the end, it ends with a call to the "quantum part" of the QNode. This quantum part is a copy of the QNode that additionally takes in the parameter vector (called `"withparams"`).

I'm not attached to these names and would be very happy to hear suggestions for better ones.

Reply:
I would put @dime10 in the loop for the names, maybe `classical_preprocessing`. I don't have a strong opinion.

Reply:
Hmm yeah no strong suggestions, although one thing that always happened was I forgot what `withparams` meant 🤔 

Some solutions might be:
- a `fn.classical` / `fn.quantum` split
- or `fn.preprocess` / `fn.quantum`
- or at the very least `fn.withgateparams` 😅

Reply:
Alright, I updated it to be `fn.preprocess` and `fn.quantum` 👍🏽